src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SocketConnection.cs

```
@@ -92,13 +92,27 @@ protected string RemoteEndpointAddressString
 
         protected static void OnReceiveTimeout(SocketConnection socketConnection)
         {
-            socketConnection.Abort(SR.Format(SR.SocketAbortedReceiveTimedOut, socketConnection._receiveTimeout), TransferOperation.Read);
+            try
+            {
+                socketConnection.Abort(SR.Format(SR.SocketAbortedReceiveTimedOut, socketConnection._receiveTimeout), TransferOperation.Read);
+            }
+            catch (SocketException)
+            {
+                // Guard against unhandled SocketException in timer callbacks
```

So you just swallow the exception?